core/sql/qmscommon/QRLogger.cpp

```diff
@@ -186,6 +188,9 @@ NABoolean QRLogger::initLog4cxx(const char* configFileName)
 {
   NAString logFileName;
 
+  if (gv_QRLoggerInitialized_)
+     return TRUE;
+ 
   // get the log directory
```

Comment:
I cannot figure out if this code is thread-safe. What if two threads concurrently enter this function and both see gv_QRLoggerInitialized_ as FALSE. Both would then call CommonLogger::initLog4cxx and possibly introduceSelf(). Is there any harm in this?